api.lua

```diff
@@ -117,18 +124,22 @@ app:match('user', '/users/:username', respond_to({
             yield_error('User ' .. self.params.username .. ' already exists');
         end
 
-        local salt = tostring(math.random())
-
         Users:create({
             created = db.format_date(),
             username = self.params.username,
-            salt = salt,
+            salt = secure_salt(),
             password = hash_password(self.params.password, salt), -- see validation.lua >> hash_password
             email = self.params.email,
             isadmin = false,
             joined = db.format_date()
         })
 
+        send_email(self.params.email, 'Welcome to Snap!', [[
+            Hello,
+
+            Welcome to Snap!
+        ]])
+
```

Comment:
Hmm, I'm guessing these are just tests?

Reply:
This one's a test yes, but that's the proper spot where we'd send an email.

I've deleted all the other debugging.